### DISCUSSION TITLE
add @global Annotation For Global Variable

### DISCUSSION THREAD

**Discussion by iqbal70** [godotengine/godot-proposals#9999] [Scripting]
Can We Add @global annotation To Make The Variable Global? Is There Any Problem With This?

**Comment by dalexeev** [comment]
Related:

* #4740
* godotengine/godot#88320

Currently you can use static variables. However, when accessing from outside the class, you need to prefix it with the class name.

> Is There Any Problem With This?

Global variables must be accessible from everywhere, so they must be resolved first. This is difficult if global variables are scattered across many scripts. Either you must add each global variable in Project Settings (like autoloads), or a script `global.gd` must be registered as global in the settings.

**Comment by Jesusemora** [comment]
We have autoloads
